src/test/java/org/dita/dost/module/BranchFilterModuleTest.java

```diff
@@ -320,4 +321,63 @@ public void testDuplicateTopic() throws IOException, SAXException {
         return res;
     }
 
+	@Test
+	public void testRewriteReferencesContainingAnchors() throws IOException, SAXException {
+	    final BranchFilterModule m = new BranchFilterModule();
+	    final Job job = new Job(tempDir,  new StreamStore(tempDir, new XMLUtils()));
+	    job.setInputDir(tempDir.toURI());
+	    final Set<Job.FileInfo> res = new HashSet<>();
+        res.add(new Job.FileInfo.Builder()
+                .src(new File(tempDir, "topicAnchors.ditamap").toURI())
+                .result(new File(tempDir, "topicAnchors.ditamap").toURI())
+                .uri(URI.create("topicAnchors.ditamap"))
+                .format(ATTR_FORMAT_VALUE_DITAMAP)
+                .build());
+        for (final String uri: Arrays.asList("filter_A.ditaval", "filter_B.ditaval")) {
+            res.add(new Job.FileInfo.Builder()
+                    .src(new File(tempDir, uri).toURI())
+                    .result(new File(tempDir, uri).toURI())
+                    .uri(URI.create(uri))
+                    .format(ATTR_FORMAT_VALUE_DITAVAL)
+                    .build());
+        }
+        for (final String uri: Arrays.asList("topic.dita")) {
+            res.add(new Job.FileInfo.Builder()
+                    .src(new File(tempDir, uri).toURI())
+                    .result(new File(tempDir, uri).toURI())
+                    .uri(URI.create(uri))
+                    .format(ATTR_FORMAT_VALUE_DITA)
+                    .build());
+        }
+        job.addAll(res);
+	    m.setJob(job);
+	    final CachingLogger logger = new CachingLogger();
+	    m.setLogger(logger);
+	    m.setXmlUtils(new XMLUtils());
+	    
+	    m.processMap(URI.create("topicAnchors.ditamap")); 
+	    assertEquals("Should re-write references", "<?xml version=\"1.0\" encoding=\"UTF-8\"?><map xmlns:ditaarch=\"http://dita.oasis-open.org/architecture/2005/\" class=\"- map/map \" ditaarch:DITAArchVersion=\"1.3\" domains=\"                          (topic abbrev-d)                          (topic delay-d)                          (map ditavalref-d)                          (map glossref-d)                          (topic hazard-d)                          (topic hi-d)                          (topic indexing-d)                          (map mapgroup-d)                          (topic markup-d xml-d)                          (topic markup-d)                          (topic pr-d)                          (topic relmgmt-d)                          (topic sw-d)                          (topic ui-d)                          (topic ut-d)                          a(props deliveryTarget)\">    \n" + 
```

Comment:
`assertEquals` is not a stable solution here. This should use XML comparison assert. Also the expected XML should not be part of the Java code, it should be a file in `src/test/resources/BranchFilterModuleTest/exp/topicAnchors.ditamap`